src/package/store.rs

```diff
@@ -58,6 +58,9 @@ impl PackageStore {
     /// Open given directory.
     pub async fn open(path: &Path) -> miette::Result<Self> {
         let store = Self::new(path.into());
+        let manifest = Manifest::read().await?;
+
+        store.populate(&manifest).await?;
```

Comment:
@mara-schulke I do the sync now here so that it happens every time the store is accessed. Not sure about the manifest however... Is it fine to read it here? Or should it be passed in (seems like touching a lot of signatures)?

Reply:
Or can we even have the manifest as member of the store?

Reply:
I think that that *could* be sensible - but i would pass it as a function argument for now, as having two instances of the manifest can become problematic in terms of state management (ie. when a dependency is added in first but not the second one)

Reply:
My gut feeling is that you want to explicitly populate the store, instead of doing it simultaneously with opening

Reply:
Just to add my 3 cents (inflation): 

I also think that with the state of the code base right now, this could be sensible. However it is not clean because we are violating some rules here:

- Naming: An `open` function does not suggest that it also performs a mutable action.
- Abstraction: We are violating abstraction layers here.

This is not really your fault — we have been violating abstraction layers before. In an ideal world, we would need three types here:

- `LocalPackage`, which owns the files for the current package (`proto`). It has methods to list, read, write files.
- `PackageStore`, which owns the files in the `proto/vendor`. It has methods to list, read, write files.
- `Manifest`, which owns the manifest file.

None of the methods of `PackageStore` should take a `Manifest`, it should not even know that manifests exist. Currently it does, because we are doing quite a bit too much in the `PackageStore`. If you think about abstractions, these low-level types all just handle one particular aspect, but this population action is something that moves things from a `LocalPackage` into a `PackageStore`, so it has to live on a higher abstraction level. So we need some kind of higher-level struct that represents the current Buffrs project:

```rust
pub struct BuffrsProject {
    pub manifest: Manifest,
    pub package: PackageStore,
    pub vendor: VendorStore
}
```

This type would have a `populate()` function that does the right thing. Any and all operations which span multiple parts should be cleanly implemented here, and the underlying types (`PackageStore`, `VendorStore`) should expose methods for interacting with the data (so we don't do `tokio::fs::write(project.package.path(), &[..])` but rather `project.package.write_write("filename", &[...])`).

 I think that right now, it is hard to do this cleanly because the PackageStore violates abstraction layers — it is supposed to only handle interactions with the `proto` folder, but it also has methods that take and use Manifests.
I think if it works the way you are doing it, it should be okay to do it that way. But at some point we should revisit this and see if we can't come up with a nice abstraction here that makes everyone happy.

Reply:
(sorry for resolving the comment, your comment was not auto-loaded, @xfbs !)

I absolutely see your point, but as you said would rather not do the restructuring (at least not in this PR). https://github.com/helsing-ai/buffrs/pull/167/commits/bf79e9eb78eb3921bad93a3a546101e2fa45d3a4
^ I changed here to pass in the manifest and doing the population explicitly. I think it is cleaner in terms of abstraction at the moment, but also more error prone... I am both happy with keeping it and leaving it for now...